server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -296,67 +299,97 @@ fn generate_response_requisition(
-                         option_id,
-                     },
-                 item_row: ItemRow { id: item_id, .. },
-                 requisition_row: _,
-             }| RequisitionLineRow {
-                id: uuid(),
-                requisition_id: response_requisition_id.to_string(),
-                item_link_id: item_id,
-                requested_quantity,
-                suggested_quantity,
-                available_stock_on_hand,
-                average_monthly_consumption,
-                snapshot_datetime,
-                comment: comment.clone(),
-                item_name,
-                initial_stock_on_hand_units,
-                incoming_units,
-                outgoing_units,
-                loss_in_units,
-                addition_in_units,
-                expiring_units,
-                days_out_of_stock,
-                option_id,
-                // Default
-                supply_quantity: 0.0,
-                approved_quantity: 0.0,
-                approval_comment: None,
+    let populate_price_per_unit = get_indicative_price_pref(connection)?;
+    let price_list = if populate_price_per_unit {
+        Some(get_pricing_for_items(
+            connection,
+            ItemPriceLookup {
+                item_ids: request_lines
+                    .iter()
+                    .map(|l| l.item_row.id.to_string())
+                    .collect(),
+                customer_name_id: None,
             },
-        )
-        .collect();
+        )?)
+    } else {
+        None
+    };
+
+    let mut response_lines = Vec::with_capacity(request_lines.len());
+    for (
+        i,
+        RequisitionLine {
+            requisition_line_row:
+                RequisitionLineRow {
+                    id: _,
+                    requisition_id: _,
+                    approved_quantity: _,
+                    approval_comment: _,
+                    item_link_id: _,
+                    supply_quantity: _,
+                    requested_quantity,
+                    suggested_quantity,
+                    available_stock_on_hand,
+                    average_monthly_consumption,
+                    snapshot_datetime,
+                    comment,
+                    item_name,
+                    initial_stock_on_hand_units,
+                    incoming_units,
+                    outgoing_units,
+                    loss_in_units,
+                    addition_in_units,
+                    expiring_units,
+                    days_out_of_stock,
+                    option_id,
+                    price_per_unit,
+                },
+            item_row: ItemRow { id: item_id, .. },
+            requisition_row: _,
+        },
+    ) in request_lines.into_iter().enumerate()
+    {
+        let price_per_unit = {
+            if price_per_unit.is_none() {
+                if let Some(price_list) = &price_list {
+                    price_list[i].calculated_price_per_unit
```

Comment:
Should be safe to index in like this with the current implementation of `get_pricing_for_items`. If we optimise `get_pricing_for_items` we'll may want to be careful that the order is maintained.